doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,198 @@
+5. Create CMakeLists.txt file in the my-kaa-application directory. It is a top-level cmake file which is responsible for the application build.
 
-<!-- TODO: Fix this link when Administration-UI-guide#AdministrationUIguide-GeneratingSDK are updated -->
-1. [Generate](Administration-UI-guide#AdministrationUIguide-GeneratingSDK) the C++ endpoint SDK in Admin UI.
-2. Download and untar the Kaa C++ SDK archive.
+        touch CMakeLists.txt
 
-   ~~~ bash
-   $ tar xfv kaa-cpp-ep-sdk.tar.gz
-   ~~~
+6. Add the following code to top-level CMakeLists.txt:
 
-3. Run the following commands.
+        cmake_minimum_required(VERSION 2.8.8)
+        project(kaa-application C)
 
-   ~~~
-   mkdir build
-   cd build
-   cmake ..
-   make
-   make install
-   ~~~
+        find_package(OpenSSL REQUIRED)
 
-## Configure C++ endpoint SDK
+        set(APP_NAME "kaa-app")
+        set(KAA_SDK_DIR "kaa")
 
-To configure the C++ endpoint SDK build, you can optionally specify the following parameters for the **cmake** command.
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
 
-{% include csv-table.html csv_path='cmake-cpp-options.csv' table_id='table-cpp-cmake-options' width='100%' %}
+        set(APPLICATION_SOURCE_FILES src/kaa-application.c)
 
-The following example illustrates the build procedure for the debug build, with the INFO log level and disabled EVENTS feature and specified path to the folder Kaa will be installed in:
+        add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/${KAA_SDK_DIR} ${CMAKE_CURRENT_BINARY_DIR}/c_sdk_build_directory)
 
-~~~
-mkdir build
-cd build
-cmake -DCMAKE_INSTALL_PREFIX='/home/username/kaa' -DKAA_DEBUG_ENABLED=1 -DKAA_MAX_LOG_LEVEL=4 -DKAA_WITHOUT_EVENTS=1 ..
-make
-make install
-~~~
+        add_executable(${APP_NAME} ${APPLICATION_SOURCE_FILES})
+        target_link_libraries(${APP_NAME} kaac crypto)
 
-# Quick way to build C/C++ endpoint SDK
+    The first two lines of the snippet set minimum cmake version that is required to build kaa-application project. Version 2.8.8 is used here because kaa-application depends on C SDK and it requires minimum CMake version to be 2.8.8. The second line sets project name and the language of the project. `find_package(OpenSSL REQUIRED)` says that OpenSSL is required to build the application, so make sure that [it is installed](#dependencies). The next two lines set the name of the executable which is built and the name of the directory which contains C SDK source files. Then some compilers flags are set and C SDK project is included as a subproject to the kaa-application project. This is a clue. The first parameter contains a path to the C SDK's CMakeLists.txt and the second one sets C SDK's build directory. `add_executable(${APP_NAME} ${APPLICATION_SOURCE_FILES})` and `target_link_libraries(${APP_NAME} kaac crypto)` compile executable named `${APP_NAME}` using `${APPLICATION_SOURCE_FILES}` and link it with `kaac` and `crypto`.
```

Comment:
Too long as for guide. Intermix cmake code with little explanation what it does.